#18 base building

jetenergy opened the pull request:
Added some more base building elements

jetenergy committed:
Cleaned up a whole bunch and fixed split bug on unload scene

jetenergy committed:
Name changes and map save

jetenergy committed:
Added base building tasks
Refactored a whole bunch of things related to Astar
Expanded the base

jetenergy committed:
Updated building queue to allow taks to be added which are not able to be done

jetenergy committed:
First pass to update level, broke the ant AI's

jetenergy committed:
saved level

jetenergy committed:
Removed feature that wasnt supposed to be here

jetenergy committed:
Hard coded teleporter fixed

MMCQ33 committed:
Merge branch 'Develop' into Level_1Updates

jetenergy committed:
Added highlighting for rooms in the queue

jetenergy committed:
Fix for instantiation errors with navmeshagent

MMCQ33 committed:
Fixed level 1 map

fixed level 1 map and the agent bugs.

MMCQ33 committed:
removed space

removed a space

jetenergy committed:
Merge branch 'Level_1Updates' into #18-BaseBuilding

# Conflicts:
#	Hivemind/Assets/Scripts/BaseBuilding/BaseController.cs

jetenergy committed:
Merge branch 'Develop' into #18-BaseBuilding

# Conflicts:
#	Hivemind/Assets/Scenes/Map.unity

jetenergy committed:
Changed scale and tried fixing the leaving base issue but i strongly think its a gathering behaviour thing

MMCQ33 reviewed (commented):
highlight different color when indestructable or unbuildable.

jetenergy committed:
Last second map save...

jetenergy committed:
Added unbuildable feedback and moved resources properly instead of constant float

jetenergy committed:
Save after close...

jetenergy committed:
Merge branch 'Develop' into #18-BaseBuilding

jetenergy committed:
Added button highlighting

jetenergy committed:
Destroy tool renamed to DestroyRoom
Renamed tool buttons on UI
Added invisible pause menu blockers so base building cant happen when paused
fixed destroy highlight
fixed worker room build queue bug

jetenergy committed:
Added job sorting for closest first

jetenergy committed:
Fixed ui button colors
